ingest/src/test/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClientTest.java

```diff
@@ -0,0 +1,352 @@
+        ingestionProperties = new IngestionProperties("dbName", "tableName");
+        ingestionProperties.setIngestionMapping("mappingName", IngestionMapping.IngestionMappingKind.Json);
+    }
+
+
+    //todo: since the blob method just forwards to the queued client I copy-pasted the methods from there, again we can
+    //      share this code, but do we want to?
+    @Test
+    void IngestFromBlob_IngestionReportMethodIsNotTable_EmptyIngestionStatus() throws Exception {
+        BlobSourceInfo blobSourceInfo = new BlobSourceInfo("http://blobPath.com", 100);
+        IngestionResult result = managedStreamingIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
+        assert result.getIngestionStatusesLength() == 0;
+    }
+
+    @Test
+    void IngestFromBlob_IngestionReportMethodIsTable_NotEmptyIngestionStatus() throws Exception {
+        BlobSourceInfo blobSourceInfo = new BlobSourceInfo("http://blobPath.com", 100);
+        ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.Table);
+
+        IngestionResult result = managedStreamingIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
+        assert result.getIngestionStatusesLength() != 0;
+    }
+
+    @Test
+    void IngestFromBlob_NullIngestionProperties_IllegalArgumentException() {
+        BlobSourceInfo blobSourceInfo = new BlobSourceInfo("http://blobPath.com", 100);
+        assertThrows(
+                IllegalArgumentException.class,
+                () -> managedStreamingIngestClient.ingestFromBlob(blobSourceInfo, null));
+    }
+
+    @Test
+    void IngestFromBlob_NullBlobSourceInfo_IllegalArgumentException() {
+        assertThrows(
+                IllegalArgumentException.class,
+                () -> managedStreamingIngestClient.ingestFromBlob(null, ingestionProperties));
+    }
+
+    @Test
+    void IngestFromBlob_IngestionReportMethodIsTable_RemovesSecrets() throws Exception {
+        BlobSourceInfo blobSourceInfo = new BlobSourceInfo("https://storage.table.core.windows.net/ingestionsstatus20190505?sv=2018-03-28&tn=ingestionsstatus20190505&sig=anAusomeSecret%2FK024xNydFzT%2B2cCE%2BA2S8Y6U%3D&st=2019-05-05T09%3A00%3A31Z&se=2019-05-09T10%3A00%3A31Z&sp=raud", 100);
+        ingestionProperties.setReportMethod(IngestionProperties.IngestionReportMethod.Table);
+        ArgumentCaptor<TableServiceEntity> captur = ArgumentCaptor.forClass(TableServiceEntity.class);
+
+        managedStreamingIngestClient.ingestFromBlob(blobSourceInfo, ingestionProperties);
+
+        verify(azureStorageClientMock, atLeast(1)).azureTableInsertEntity(anyString(), captur.capture());
+        assert (((IngestionStatus) captur.getValue()).getIngestionSourcePath()).equals("https://storage.table.core.windows.net/ingestionsstatus20190505");
+    }
+
+    //todo: all of these null/empty argument checks are also repetitive, but maybe that's just java
```

Comment:
Please change to  -> // TODO

Its not really repetitive - its just easier for debugging as separate methods but it doesn't have to be - the asserts are one liners ..